common-rest/src/main/java/org/uniprot/api/rest/service/query/processor/UniProtSpecialCharacterQueryNodeProcessor.java

```diff
@@ -0,0 +1,157 @@
+package org.uniprot.api.rest.service.query.processor;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.CharSequenceReader;
+import org.apache.lucene.analysis.charfilter.MappingCharFilter;
+import org.apache.lucene.analysis.charfilter.NormalizeCharMap;
+import org.apache.lucene.queryparser.flexible.core.QueryNodeException;
+import org.apache.lucene.queryparser.flexible.core.nodes.FieldQueryNode;
+import org.apache.lucene.queryparser.flexible.core.nodes.QueryNode;
+import org.apache.lucene.queryparser.flexible.core.processors.QueryNodeProcessorImpl;
+
+import java.io.Reader;
+import java.util.List;
+
+public class UniProtSpecialCharacterQueryNodeProcessor extends QueryNodeProcessorImpl {
+    @Override
+    protected QueryNode preProcessNode(QueryNode node) {
+        return node;
+    }
+
+    @Override
+    protected QueryNode postProcessNode(QueryNode node) throws QueryNodeException {
+        if (node.isLeaf() && node instanceof FieldQueryNode fieldQueryNode) {
+            try (UniProtSpecialCharacterFilter uniProtSpecialCharacterFilter =
+                         new UniProtSpecialCharacterFilter(new CharSequenceReader(fieldQueryNode.getText()))) {
+                uniProtSpecialCharacterFilter.read();
+                uniProtSpecialCharacterFilter.reset();
+                fieldQueryNode.setText(IOUtils.toString(uniProtSpecialCharacterFilter));
+            } catch (Exception e) {
+                throw new QueryNodeException(e);
```

Comment:
We should catch specific exception IOEXception instead of broader one Exception. 
Could you please add a test for the exception case ? What happens to the search and search query in case of exception?